libdrgn/drgn.h

```diff
@@ -2983,6 +2983,44 @@ enum drgn_symbol_kind {
 	DRGN_SYMBOL_KIND_IFUNC = 10, /* STT_GNU_IFUNC */
 };
 
+/** Describes the lifetime of an object provided to drgn */
+enum drgn_lifetime {
```

Comment:
:heart: :heart: :heart: This approach is way cleaner and more generic than I expected. Great job.

Reply:
Thanks!